contrib/platform/src/com/sun/jna/platform/win32/WTypes.java

```diff
@@ -160,15 +164,29 @@ public BSTRByReference(BSTR value) {
         }
 
         public void setValue(BSTR value) {
+            // Save a local reference to prevent losing value's memory
+            this.bstr = value;
             this.getPointer().setPointer(0, value.getPointer());
         }
 
         public BSTR getValue() {
-            return new BSTR(getPointer().getPointer(0));
+            // If memory address hasn't changed, return the stored BSTR
+            if (bstr != null) {
+                if (getPointer().getPointer(0).equals(bstr.getPointer())) {
```

Comment:
The null check could succeed and after that bstr is updated from a parallel thread and set to null. I suggest to copy the value into a local variable, read that and update bstr if necessary.

Reply:
True but this is starting to get to be too much work for a band-aid fix to a poor API to start with.  I'm really leaning more towards a simple deprecation.